Use payment methods api instead of sources

reykjalin opened the pull request:
<!--
Did I add a title? A descriptive, yet concise, title.
-->

<!--
Issue: Link to the GitHub issue this PR addresses (if appropriate).
-->

Fixes N/A

## Changes proposed in this Pull Request:

<!--
Description: Write a brief summary about this PR. As you compose your summary, consider each of these questions and address them if appropriate. Why is this change needed? What does this change do? Were there other solutions you considered? Why did you choose to pursue this solution? Describe any trade-offs you might have had to make.
-->

<!--
Questions for the PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

<!--
Images or gifs: Include before and after screenshots or gifs/videos when it makes sense.
-->

- Use the PaymentMethods API instead of Sources while maintaining compatibility with Orders that may have used Sources in the past.

This is a prerequisite for supporting the Indian eMandates for recurring payments (see #2561 ).

## Testing instructions

All critical flows should pass.

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Please follow the following guidelines when writing testing instructions:

- Include screenshots if there is no similar flow in the critical flows: https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Critical-flows
- Assume instructions will be copied over to the Release Testing Instructions wiki page: https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions
- Assume instructions will be followed by external testers.
- Assume tester does not have intimate knowledge of Stripe.
-->

### Payments with UPE-saved cards (`pm_abc123`) should work with UPE disabled without going through the UPE gateway

1. Save card with UPE active.
2. Turn off UPE.
3. Checkout with the saved card.
4. Payment should be successful.


### Payments with sources (`src_abc123`) should still work

1. Save a card with UPE disabled on `develop`.
2. Switch to this branch `git switch add/use-payment-methods-api-instead-of-sources`.
3. Checkout with the saved card.
4. Payment should be successful.

---

-   [ ] Covered with tests (or have a good reason not to test in description ☝️)
-   [ ] Added changelog entry **in both** `changelog.txt` and `readme.txt` (or does not apply)
-   [ ] Tested on mobile (or does not apply)

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)

Commit:
add helper function for retrieveing payment methods and sources

Commit:
use helper function to retrieve sources

dougaitken commented:
Dropping a few possible issues that might get "fixed" with this one @reykjalin 

Stop saving new cards to sources. Use the PaymentMethods API instead #1424
Use payment_method instead of source for requests to the Stripe API #1486 
Use Payments API for adding/deleting saved cards (an SEPA Debit) in my-account and when making a payment #1614

I think there's others but that's a start

reykjalin commented:
Awesome, thank you Doug!

Commit:
remove references to sources from JS

Commit:
Make PHP mostly compatible with JS changes

Commit:
Refer to payment methods instead of sources

Except where we're explicitly working with only sources as opposed to
both sources and other types of payment methods.
